truffle/test/Splitter.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const Splitter = artifacts.require('Splitter');
+const truffleAssert = require('truffle-assertions');
+const BN = web3.utils.BN;
+
+contract('Splitter', (accounts) => {
+    let instance;
+
+    // Initiating new contract before each test
+    beforeEach("Creating new contract", async () => {
+        instance = await Splitter.new({from: accounts[0]});
+    });
+
+    //Testing the balance update
+    it('Balance should be equal to 1 ether', () => {
+        return instance.splitEth(accounts[1], accounts[2], {from: accounts[0], value: web3.utils.toWei("1", "ether")})
+            .then(txObj => {
+                assert.strictEqual(txObj.logs.length, 1);
+                assert.strictEqual(txObj.logs[0].event, "LogEthSent");
+
+                return web3.eth.getBalance(instance.address);
+            })
+            .then(balance => {
+                assert.strictEqual(balance, web3.utils.toWei("1", "ether"));
+            });
+    });
+
+    // Testing to split ether if the contract has been paused
+    it('The splitEth function should not be usable when the contract is paused', async () => {
+        const txObj = await instance.pauseContract();
+        assert.strictEqual(txObj.logs.length, 1);
+        assert.strictEqual(txObj.logs[0].event, "LogPausedContract");
+
+        await truffleAssert.reverts(
+            instance.splitEth(accounts[1], accounts[2], {from: accounts[0], value: web3.utils.toWei("10", "ether")})
+        );
+    });
+
+    // Testing the withdraw function with empty balance
+    it('Withdraw should fail on empty balance', async () => {
+        await truffleAssert.reverts(
+            instance.withdraw({from: accounts[1]})
+        );
+    });
+
+    // Checking the final balance with even number
+    it('Balances should be emptied on even number', () => {
+        return instance.splitEth(accounts[1], accounts[2], {from: accounts[0], value: web3.utils.toWei("3", "ether")})
```

Comment:
Perhaps a `describe("withdraw"`, and inside it, a `beforeEach("make a split"`?

Reply:
I don't see the purpose of this. Indeed, I only have two tests that make a split before withdrawing on these two tests don't send the same amount of ether.

Reply:
True, that precludes this `beforeEach`.